camayoc/tests/qcs/api/v1/scanjobs/test_run_scanjobs.py

```diff
@@ -303,3 +314,76 @@ def test_scan_task_results(scan_info):
         num_failed = task['systems_failed']
         num_scanned = task['systems_scanned']
         assert num_scanned == sys_count - num_failed
+
+
+def test_disabled_optional_products_facts():
+    """Test scan jobs from scans with disabled optional products.
+
+    :id: 6f91ea5c-32b9-11e8-b467-0ed5f89f718b
+    :description: Test that scan jobs of scans with disabled products
+        are not gathering the raw_facts defined in the roles of the
+        disabled products.
+    :steps:
+        1) Iterate over the scans and gather the disabled_optional_products
+        2) For any scan that was defined with disabled_optional_products:
+            a) Create a facts_to_ignore list composed of raw_facts for each
+                disabled role
+            b) Iterate through the inspection results fact dictionaries
+            c) Assert that no facts that should be ignored are in the
+                dictionaries
+    :expectedresults: No facts are collected for disabled products
+    """
+    FOUND = False
+    scans = scan_info()
+    facts_to_ignore = []
+    for scan in scans:
+        disabled_optional_products = scan.get('disabled_optional_products')
+        if disabled_optional_products:
+            # Build the list of facts that should not be in inspection results
+            for product in disabled_optional_products.keys():
+                if product == 'jboss_eap':
+                    facts_to_ignore += QCS_EAP_RAW_FACTS
+                elif product == 'jboss_fuse':
+                    facts_to_ignore += QCS_FUSE_RAW_FACTS
+                elif product == 'jboss_brms':
+                    facts_to_ignore += QCS_BRMS_RAW_FACTS
+            # grab the inspection results of the scan
+            inspection_results = \
+                get_scan_result(scan.get('name')).get('inspection_results')
+            for system in inspection_results:
+                fact_dicts = system.get('facts')
+                # grab the facts for each system
+                for dictionary in fact_dicts:
+                    for fact in facts_to_ignore:
+                        if fact in dictionary.values():
+                            FOUND = True
+        assert not FOUND
+
+
+def test_disabled_optional_products():
+    """Test scan jobs from scans with disabled_optional_products.
+
```

Comment:
You'll have to tweak the test some but you can get it so you only get one scan at a time by parametrizing the test on the scans. You do this by changing the signature to this:

```diff
@pytest.mark.parametrize(
    'scan_info', scan_info(), ids=utils.name_getter)
+def test_disabled_optional_products(scan_info):
+    """Test scan jobs from scans with disabled_optional_products.
```

Then instead of `for scan in scans ...` you would just have one scan by calling `scan = get_scan_result(scan_info['name'])`

Sorry so many things named so similarly. Any suggestions welcome